spec/formatting.md

```diff
@@ -928,7 +928,8 @@ Implementations MAY provide other _bidirectional isolation strategies_.
 
 Implementations MAY supply a _bidirectional isolation strategy_ that performs no processing.
 
-The _Default Bidi Strategy_ is defined as follows:
+The _Default Bidi Strategy_ is defined as a function `B` from expressions
```

Comment:
Linkify _expressions_.

This is weird? Where does `B` come from? Why is it a "function"? The text above this talks about "bidirectional isolation strategies". I think this needs to be different.

Perhaps:

> The _Default Bidi Strategy_ is defined as follows:
>
> 1. Let `out` be an empty string to contain the formatted output of the _message_
> 1. Let `msgdir`...
> 1. For each _expression_ `exp` in the _pattern_:
>    1. If `exp` is a plain literal (text), append `exp` to `out`.
>    1. If `exp` is an _placeholder_, let `fmt` be the formatted string representation of
>       the _resolved value_ of `exp`.
>    1. Let `dir` be the directionality of `fmt`...
>     // etc.
>     1. Append `fmt` to `out`.
> 1. Emit `out` as the formatted output of the _message_.

Note that we don't have a name for the non-placeholder bits of text in a pattern. I call them expressions here, but they aren't really expressions.

It would be even better if the section on formatting just above here defined how to make the string output. The "bidi strategy" then could be reference in that set of instructions (_"apply the appropriate bidirectional isolation strategy to the contents of expression"_)

Reply:
I rewrote this imperatively instead of as a function. I also did a rewrite to call the components of a pattern a "part", and distinguish between text parts and placeholder parts.